marimo/_messaging/errors.py

```diff
@@ -107,6 +107,50 @@ def describe(self) -> str:
         return self.msg
 
 
+@dataclass
+class MarimoInternalError:
+    """
+    An internal error that should be hidden from the user.
+    The error is logged to the console and then a new error is broadcasted
+    such that the data is hidden.
+
+    They can be linked back to the original error by the error_id.
+    """
+
+    error_id: str
+    type: Literal["internal"] = "internal"
+    msg: str = ""
+
+    def __post_init__(self) -> None:
+        self.msg = f"An internal error occurred: {self.error_id}"
+
+    def describe(self) -> str:
+        return self.msg
+
+
+def is_unexpected_error(error: Error) -> bool:
+    """
+    These errors are unexpected, in that they are not intentional.
+    mo.stop and interrupt are intentional.
+    """
+    return error.type not in [
+        "ancestor-prevented",
+        "ancestor-stopped",
+        "interruption",
+    ]
+
+
+def is_sensitive_error(error: Error) -> bool:
```

Comment:
This seems to be the same as `is_unexpected_error`?

Reply:
They are slightly different. One is interrupt and the other is internal